packages/experimental/Avatar/ios/MSFAvatarViewManager.m

```diff
@@ -1,82 +1,105 @@
-#import <React/RCTComponent.h>
 #import <React/RCTViewManager.h>
 
+#import "MSFAvatarStorage.h"
+
 @import FluentUI;
 
-@implementation RCTConvert (MSFAvatarViewAdditions)
+// Macros inspired by https://betterprogramming.pub/react-native-meets-swiftui-d1606a8e1681
+
+#define RCT_EXPORT_SWIFTUI_PROPERTY(name, type, proxyClass) \
+RCT_REMAP_VIEW_PROPERTY(name, __custom__, type)  \
+- (void)set_##name:(id)json forView:(UIView *)view withDefaultView:(UIView *)defaultView RCT_DYNAMIC { \
+    NSMutableDictionary *storage = [proxyClass storage]; \
+    proxyClass *proxy = storage[[NSValue valueWithNonretainedObject:view]];  \
+    proxy.state.name = [RCTConvert type:json]; \
+}
+
+#define RCT_REMAP_SWIFTUI_PROPERTY(name, keyPath, type, proxyClass) \
+RCT_REMAP_VIEW_PROPERTY(name, __custom__, type)  \
+- (void)set_##name:(id)json forView:(UIView *)view withDefaultView:(UIView *)defaultView RCT_DYNAMIC { \
+    NSMutableDictionary *storage = [proxyClass storage]; \
+    proxyClass *proxy = storage[[NSValue valueWithNonretainedObject:view]];  \
+    proxy.state.keyPath = [RCTConvert type:json]; \
+}
+
+#define RCT_EXPORT_CUSTOM_SWIFTUI_PROPERTY(name, type, proxyClass) \
+RCT_REMAP_VIEW_PROPERTY(name, __custom__, type)  \
+- (void)set_##name:(id)json forView:(UIView *)view withDefaultView:(UIView *)defaultView RCT_DYNAMIC
+
+
```

Comment:
nit- Probably don't need this many empty lines between the API and the next line?